src/main/java/org/wso2/lsp4intellij/client/languageserver/requestmanager/DefaultRequestManager.java

```diff
@@ -397,8 +398,12 @@ public void didClose(DidCloseTextDocumentParams params) {
     public CompletableFuture<CompletionItem> resolveCompletionItem(CompletionItem unresolved) {
         if (checkStatus()) {
             try {
-                return (serverCapabilities.getCompletionProvider() != null && serverCapabilities.getCompletionProvider()
-                        .getResolveProvider()) ? textDocumentService.resolveCompletionItem(unresolved) : null;
+                return (
+                    Optional.of(serverCapabilities.getCompletionProvider())
+                        .map(x -> x.getResolveProvider())
```

Comment:
`.map(CompletionProvider::getResolveProvider)`

and so on... @NipunaRanasinghe WDYT?

Reply:
I intentionally used a lambda instead as it reads a bit easier to me. Doesn't seem valuable to make things mire verbose than necessary IMHO.

Reply:
`::` syntax also provides info about type of `x`, but I understand your point.

Reply:
> `.map(CompletionProvider::getResolveProvider)`
> 
> and so on... @NipunaRanasinghe WDYT?

@nixel2007 @matthewmichihara I'm being 0 here as the 2 approaches don't have a performance different so it's just a matter of readability + personal preference. :)